modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/dto/v2_0/util/DataLayoutUtil.java

```diff
@@ -191,6 +236,55 @@ private static DataLayoutRow _toDataLayoutRow(
 		);
 	}
 
+	private static DataRule[] _toDataRules(
+		List<DDMFormRule> ddmFormRules,
+		SPIDDMFormRuleConverter spiDDMFormRuleConverter) {
+
+		DataRule[] dataRules = new DataRule[0];
+
+		for (SPIDDMFormRule spiDDMFormRule :
+				spiDDMFormRuleConverter.convert(ddmFormRules)) {
+
+			DataRule dataRule = new DataRule();
+
+			Gson gson = new Gson();
+
+			dataRule.setActions(
+				Stream.of(
+					spiDDMFormRule.getSPIDDMFormRuleActions()
+				).flatMap(
+					Collection::stream
+				).map(
+					spiDDMFormRuleAction -> gson.fromJson(
+						JSONFactoryUtil.looseSerializeDeep(
+							spiDDMFormRuleAction),
+						Map.class)
+				).toArray(
+					Map[]::new
+				));
+
+			dataRule.setConditions(
+				Stream.of(
+					spiDDMFormRule.getSPIDDMFormRuleConditions()
+				).flatMap(
+					Collection::stream
+				).map(
+					spiDDMFormRuleCondition -> gson.fromJson(
+						JSONFactoryUtil.looseSerializeDeep(
+							spiDDMFormRuleCondition),
+						Map.class)
+				).toArray(
+					Map[]::new
+				));
+
+			dataRule.setLogicalOperator(spiDDMFormRule.getLogicalOperator());
+
+			dataRules = ArrayUtil.append(dataRules, dataRule);
```

Comment:
why append and not just

```suggestion
			dataRules.add(dataRule);
```